datafusion/physical-plan/src/execution_plan.rs

```diff
@@ -424,6 +433,17 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     }
 }
 
+#[derive(Clone, Copy)]
+pub enum InvariantLevel {
```

Comment:
I think the fact this has the same name as the [logical `InvariantLevel`](https://github.com/apache/datafusion/blob/0283077c63d7b9a7464bc75f0686774cd5d10924/datafusion/expr/src/logical_plan/invariants.rs#L31) may be confusing

However, calling it `ExecutionInvariantLevel` would make the names inconsistent 🤔 

Maybe we can just add a few comments to the struct like this

```suggestion
/// [`ExecutionPlan`] Invariant Level
///
/// What set of assertions ([Invariant]s)  holds for a particular `ExecutionPlan`
///
/// [Invariant]: https://en.wikipedia.org/wiki/Invariant_(mathematics)#Invariants_in_computer_science
#[derive(Clone, Copy)]
pub enum InvariantLevel {
```